gcetcbendorsement/cmd/root.go

```diff
@@ -96,15 +97,33 @@ func (b *bearerGetter) Get(url string) ([]byte, error) {
 	return io.ReadAll(resp.Body)
 }
 
+// The go-tdx-guest IsSupported signature is different from go-sev-guest. Add a compatibility layer.
+type tdxQuoteProvider struct{ p tdclient.QuoteProvider }
+
+func (qp *tdxQuoteProvider) IsSupported() bool { return qp.p.IsSupported() == nil }
+func (qp *tdxQuoteProvider) GetRawQuote(reportData [64]byte) ([]uint8, error) {
+	return qp.p.GetRawQuote(reportData)
+}
+
+func getProvider() (p extract.QuoteProvider) {
+	p, _ = client.GetQuoteProvider()
+	if p.IsSupported() {
+		return p
+	}
+	tdp, _ := tdclient.GetQuoteProvider()
+	p = &tdxQuoteProvider{p: tdp}
+	if !p.IsSupported() {
+		logger.Warningf("Could not create a supported quote provider.")
```

Comment:
Previous version of the code had a fatal error. Did you check that everything is OK with a nil provider?

The previous version also checked for an error. Is that impossible now? Will GetQuoteProvider always return a valid instance?

Reply:
It will always return an object and not error, correct. That's been moved to the IsSupported call. Just an outdated signature.

Reply:
A nil provider is indeed fine.